internal/pkg/workspace/workspace.go

```diff
@@ -279,12 +283,21 @@ func (ws *Workspace) ReadWorkloadManifest(mftDirName string) (WorkloadManifest,
 		return nil, err
 	}
 	mft := WorkloadManifest(raw)
-	mftName, err := mft.workloadName()
+	if err := ws.manifestNameMatchWithDir(mft, mftDirName); err != nil {
+		return nil, err
+	}
+	return mft, nil
+}
+
+// ReadEnvironmentManifest returns the contents of the environment's manifest under copilot/environments/{name}/manifest.yml.
+func (ws *Workspace) ReadEnvironmentManifest(mftDirName string) (WorkloadManifest, error) {
+	raw, err := ws.read(environmentsDirName, mftDirName, manifestFileName)
 	if err != nil {
 		return nil, err
 	}
-	if mftName != mftDirName {
-		return nil, fmt.Errorf(`name of the manifest "%s" and directory "%s" do not match`, mftName, mftDirName)
+	mft := WorkloadManifest(raw)
```

Comment:
This doesn't look like it should belong to here 🤔

Reply:
why? `WorkloadManifest` is just a manifest with `name` and `type` fields

Reply:
Because this is for env manifest? I would assume `WorkloadManifest` the method is for workload manifest only. I think though they have the same struct, we probably want to give the public method a different name.

Reply:
updated